Daisy Toolchain Installers

beserge opened the pull request.

beserge committed:
Simple cmake/cpack hello world test

beserge committed:
Fix missing files bug, create (nonworking) windows installer

beserge commented:
Working on getting the Windows NSIS installer working. We can use the DragNDrop generator for creating a macOS diskimg type installer as well.

beserge committed:
Add default gitignore

beserge committed:
Adjust default install dir

beserge committed:
Remove icon line, add cleanup to create script

beserge committed:
Create README with special NSIS notes

beserge committed:
Install windows toolchain

beserge committed:
Remove helloworld test

beserge committed:
Update contact email and version number

andrewikenberry commented:
- I want to remove the "Windows protected your pc" nonsense
- I want it to always add it to path, or in the very least have path be added as the default option

beserge commented:
`make` issue:

```  
Cannot find file at '..\\lib\make\tools\install\bin\make.exe'
(C:\Program Files\DaisyToolchain\lib\make\tools\install\bin\make.exe). 
This usually indicates a missing or moved file.
```

beserge committed:
Custom NSIS templating, change add PATH to default

The information for how to use NSIS templates with cpack  exists nowhere on the internet.
I was able to find a git repo (opensim-gui) which does it by chance.

beserge commented:
I've made `add PATH to current user` the default option. I've also figured out how to get total control back over the NSIS files, so I basically have total power now. Meaning, if I can figure out how NSIS works, I can put anything in, take anything out of the installer that we want. 

I'm going to go about removing the start menu folder page next as Shens requested. 

As for the Windows protected your PC stuff, I've been unable to recreate it so far. I know it's caused by a particular Windows defender setting, but my PC doesn't seem to take issue with the installer. Perhaps because I'm the one who built it. 🤷 
@stephenhensley did you have that popup when running the installer? Do you have that popup when installing other programs?

beserge committed:
Comment out start menu folder page

stephenhensley commented:
> . . . so I basically have total power now.
\- Ben, 2021

@beserge @andrewikenberry I did have the same "protect my pc" stuff. I think we have to "code sign" the exe to make it so it doesn't throw that flag when launched on a PC that didn't build it.

Not exactly sure how to do it, but I imagine the mechanism is built-in to CPack. However, I imagine we'll have to get a certificate generated from msft/apple in some form or another for it to work.

beserge committed:
Fix default PATH bug

Default PATH was selected in menu, but the value wasnt being reflected in the code.
I manually set the $ADD_TO_PATH_CURRENT_USER var to 1.

beserge committed:
remove rm build/ step from install script

beserge committed:
Install entirety of necessary toolchain files

beserge committed:
Regenerate NSIS template to fix uninstaller

beserge committed:
Add more detail to Readme.md

beserge committed:
License file for installer

beserge committed:
Add license to installer

beserge committed:
Seperate scripts for creating installers on macos and windows

beserge committed:
Rm old create_installers

beserge committed:
First pass at macos custom installer

beserge committed:
Continue fleshing out tkniter installer

beserge committed:
Remove mac tools

beserge committed:
Remove unix section from cmake, add shebang to win script

beserge committed:
Move NSIS files to resource dir

beserge commented:
Final note for now:
We've given up on the mac branch of this project for now. There's some work on various aspects of that on the `mac_installer `branch for safe keeping.

The Windows installer is however ready to rock! 🎸
